src/test/java/com/iexec/worker/chain/ContributionServiceTests.java

```diff
@@ -100,26 +101,32 @@ void shouldChainTaskNotBeInitialized() {
     @Test
     void getCannotContributeStatusCauseShouldReturnWorkerpoolAuthorizationNotFound() {
         final String chainTaskId = chainTask.getChainTaskId();
+        final TaskDescription contributeAndFinalizeTaskDescription = TaskDescription.builder()
+                .chainTaskId(chainTaskId)
+                .trust(BigInteger.ONE)
+                .isTeeTask(true)
+                .build();
 
         when(workerpoolAuthorizationService.getWorkerpoolAuthorization(chainTaskId)).thenReturn(null);
+        when(iexecHubService.getTaskDescription(chainTaskId)).thenReturn(contributeAndFinalizeTaskDescription);
+        when(iexecHubService.getChainTask(chainTaskId)).thenReturn(Optional.of(chainTask));
 
-        assertThat(contributionService.getCannotContributeStatusCause(chainTaskId).orElse(null))
-                .isEqualTo(WORKERPOOL_AUTHORIZATION_NOT_FOUND);
+        List<ReplicateStatusCause> causes = contributionService.getCannotContributeStatusCause(chainTaskId);
+        assertThat(causes).containsExactly(WORKERPOOL_AUTHORIZATION_NOT_FOUND);
```

Comment:
I find this form more readable. What do you think ?
```suggestion
        assertThat(contributionService.getCannotContributeStatusCause(chainTaskId))
                .containsExactly(WORKERPOOL_AUTHORIZATION_NOT_FOUND);
```

Reply:
Resolved in [ac9313c](https://github.com/iExecBlockchainComputing/iexec-worker/pull/659/commits/ac9313cc6e7ca6bd23048a03e7f233a49c2b213f)